src/dbnode/persist/fs/files.go

```diff
@@ -1220,3 +1336,75 @@ func snapshotIndexSegmentFilePathFromTimeAndIndex(
 	suffix := filesetIndexSegmentFileSuffixFromTime(t, segmentIndex, segmentFileType)
 	return filesetPathFromTimeAndIndex(prefix, t, snapshotIndex, suffix)
 }
+
+func snapshotMetadataFilePathFromIdentifier(prefix string, id SnapshotMetadataIdentifier) string {
+	return path.Join(
+		prefix,
+		snapshotDirName,
+		fmt.Sprintf(
+			"%s-%s-%d-%s%s",
+			snapshotFilePrefix, sanitizeUUID(id.UUID.String()), id.Index, metadataFileSuffix, fileSuffix))
+}
+
+func snapshotMetadataCheckpointFilePathFromIdentifier(prefix string, id SnapshotMetadataIdentifier) string {
+	return path.Join(
+		prefix,
+		snapshotDirName,
+		fmt.Sprintf(
+			"%s-%s-%d-%s-%s%s",
+			snapshotFilePrefix, sanitizeUUID(id.UUID.String()), id.Index, metadataFileSuffix, checkpointFileSuffix, fileSuffix))
+}
+
+// sanitizeUUID replaces all instances of separator ("-") with uuidSeparator ("_") in the provided
+// uuid string. If we did not do this, then when we tried to split up filenames based on the separator
+// every "piece" of the UUID would be included as a separate fragment from the name when instead we want
+// to treat the entire UUID as one "piece" of the name.
+func sanitizeUUID(uuid string) string {
```

Comment:
Ugh.. this is pretty gross.  Can you just strip the "-" completely?  UUID lib seems to support just reading raw hex without dashes, etc (this is doc string of the func that it calls to parse itself under the hood):
```
// Parse decodes s into a UUID or returns an error. Both the standard UUID forms of xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx and urn:uuid:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx are decoded as well as the Microsoft encoding {xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx} and the raw hex encoding: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.
```

Reply:
done